src/Compilers/Core/MSBuildTask/Csc.cs

```diff
@@ -393,7 +411,7 @@ internal string GetDefineConstantsSwitch(string originalDefineConstants)
             // add them to the outgoing string.
```

Comment:
See the doc comment for this method GetDefineConstantsSwitch (GitHub does not allow me to place a comment there). It says "Note that CSharp does support assigning a value to the constants", but probably means "Note that CSharp does NOT support assigning a value to the constants".